ts/util/LinkedList.ts

```diff
@@ -224,48 +211,49 @@ export class LinkedList<DataClass> {
   }
 
   /**
-   * Make the list iterable and return the data from the items in the list
+   * Forward iterator for the list
    *
-   * @return {{next: Function}}  The object containing the iterator's next() function
+   * @yield {DataClass} The next item in the iteration sequence
    */
-  public [Symbol.iterator](): Iterator<DataClass> {
+  protected *iterator(): IterableIterator<DataClass> {
     let current = this.list;
-    return {
-                                                                    /* tslint:disable-next-line:jsdoc-require */
-      next() {
-        current = current.next;
-        return (current.data === END ?
-                {value: null, done: true} :
-                {value: current.data, done: false}) as IteratorResult<DataClass>;
-      }
-    };
+
+    while (current.data !== END) {
+      yield current.data as DataClass;
+      current = current.next;
+    }
+  }
+
+  /**
+   * Reverse iterator for the list
+   *
+   * @yield {DataClass} The previous item in the iteration sequence
+   */
+   protected *reverse_iterator(): IterableIterator<DataClass> {
```

Comment:
In MathJax, we use camelCase for function names, not underscores.  Can you change this to `reverseIterator`?

I'm also wondering if these need to be separate methods, or if they could be folded in the main functions.  E.g.:

``` javascript
 public reversed(): IterableIterator<DataClass> {
     return function * (): IterableIterator<DataClass> {
       let current = this.list;
       while (current.data !== END) {
         yield current.data as DataClass;
         current = current.prev;
       }
     }
   }
```

or even just

``` javascript
 public *reversed(): IterableIterator<DataClass> {
     let current = this.list;
     while (current.data !== END) {
       yield current.data as DataClass;
       current = current.prev;
     }
   }
```

I haven't tried these, but it seems unnecessary to have a method that just returns the result of another method.